WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift

```diff
@@ -34,6 +34,9 @@ final class ApplicationPasswordAuthorizationWebViewController: UIViewController
         return indicator
     }()
 
+    /// The view controller that was presenting the application password flow.
+    private weak var previousViewController: UIViewController?
```

Comment:
Just wondering: is there any case where `previousViewController` can be deallocated when the user is still on `ApplicationPasswordAuthorizationWebViewController`? I'm curious why the weak reference is necessary here.

Reply:
I would say that it's more of a defensive approach. Do you think `previousViewController` might be deallocated if it's no longer needed or if the navigation stack changes (e.g., the user navigates away from the flow that involves `previousViewController`)?

Reply:
My understanding is that `previousViewController` should not be deallocated since it's part of the navigation stack that contains the `ApplicationPasswordAuthorizationWebViewController`.  The reason is that when one navigates back, the view controller and its states should still be there instead of being reinitialized. WDYT?

Reply:
I re-analyzed the whole flow, I think you are right, the whole flow will be deallocated all at once. I'm going to remove `weak`